nexus-and-alo/usage-scenarios.md

```diff
@@ -0,0 +1,32 @@
+
+## Exposing non-Temporal application with Temporal-esque calling semantics (ALO)
+
+**Problem statement:** When designing and building APIs, there is often a need to expose operations which are not well suited for traditional request/response semantics. In many cases, developers simply shoehorn the arbitrary length operations into a traditional req/res approach resulting in bad dx at minimum and catastrophic bugs in the worst case. There exists no standardized approach for calling operations of arbitrary length in a reliable and consistent manner. There are schema specifics approaches such as [grpc long running](https://cloud.google.com/spanner/docs/reference/rpc/google.longrunning) and approaches that marry themselves to specific architectures like pub/sub which is the case with [Async API](https://www.asyncapi.com/). 
+
+**Solution:** While there is no open standard for exposing ALOs, there is a very reasonable Temporal-specific approach to this problem today - Workflows. Workflows provide the minimum set of APIs and functionality required to surface ALOs in a reliable and enjoyable fashion. Taking inspiration from Workflows, we are proposing a standard for ALOs which provide a Workflow-esque calling experience for any backend that supports the standard - not just Temporal. ALOs will be a first class citizen in Temporal, but any ALO compliant backend can be interacted with as if you were calling Workflows. 
+
+## Exposing non-Temporal application via Nexus
+
+**Problem statement:** Calling into external APIs is an inconsistent experience as each API developer makes different choices, adds unique bugs and provides different levels of guarantees. As a consumer looking for consistency and similarity of guarantees between APIS , this reality provides a far from ideal experience.
+
+**Solution:** While most API developers will likely not adopt Temporal or the ALO concept immediately, this is not required to provide a consistent experience for consuming miscellaneous external APIs. API consumers or producers looking to provide a more reliable experience for those calling their API can simply wrap the existing API with a Temporal Nexus. To be clear, the developer would not be required to use any other Temporal primitives (Workflows & Activities) in this scenario at all, just Nexus. Wrapping an API (even one that you don’t own) with Nexus would immediately provide significant improvements in resiliency, availability and developer experience. All calls to the API would come with automatic retries out of the box, and ALOs could be manually surfaced by connecting various endpoints of the backing API to the expected ALO methods.
```

Comment:
I get that this doc can't really get into implementation details, but, this bit in particular feels a bit handwave-y.

If it was reduced to "Nexus should be able to wrap existing APIs to provide a Temporal-workflow-like experience" I would've learned the same amount.

In particular
>ALOs could be manually surfaced by connecting various endpoints of the backing API to the expected ALO methods

Is hard to understand.

What do I get here besides auto-retries? (Even just that is pretty nice!) Maybe a brief example would be clarifying?

Reply:
There is actually an example of this already. The manual HTTP ALO implementation provided in the ALO strawman is meant to demonstrate this:

https://github.com/temporalio/proposals/blob/nexus-alo/nexus-and-alo/alo-strawman.md#a-non-temporal-example-of-alos

Reply:
Its more than just retries, basically any API which surfaces the building blocks of an ALO (resource based APIs) could be surfaced as one. 

That being said, regardless of my view on the actual solution this section is not clear enough and I will improve it. Thanks for calling that out!